lightning/src/util/enforcing_trait_impls.rs

```diff
@@ -48,7 +48,11 @@ impl EnforcingChannelKeys {
 }
 
 impl ChannelKeys for EnforcingChannelKeys {
-	fn commitment_secret(&self, idx: u64) -> [u8; 32] { self.inner.commitment_secret(idx) }
+	fn get_per_commitment_point<T: secp256k1::Signing + secp256k1::Verification>(&self, idx: u64, secp_ctx: &Secp256k1<T>) -> PublicKey {
+		self.inner.get_per_commitment_point(idx, secp_ctx)
+	}
+
+	fn revoke_commitment(&self, idx: u64) -> [u8; 32] { self.inner.revoke_commitment(idx) }
```

Comment:
Out of curiosity, do all the tests still pass if we require that we never re-sign a commitment transaction after revoking its index? Seems like thats the right next step to enforcement here.

Reply:
I pushed a commit for that, but `test_fee_spike_violation_fails_htlc` is doing something out of order, even after an attempted fix.  I can try to correct this - may need to move things around in the test.  Let me know what you think of the direction.

Reply:
I force-pushed a fix for that, should pass now.

Reply:
So the implementation is incomplete, will fix that in a bit.

Reply:
I tried to do this, but I'm seeing some persistence issues.  I think using persistent fields in `EnforcingChannelKeys` doesn't work because there are multiple copies of the class around (likely via read/write), so the `revoked_commitment` field isn't staying updated as expected.

For now, I moved the third commit and the enforcement WIP commit to a new PR in our repo: https://github.com/lightning-signer/rust-lightning/pull/1